rcl_action/include/rcl_action/goal_state_machine.h

```diff
@@ -83,23 +80,38 @@ _set_canceled_event_handler(rcl_action_goal_state_t state, rcl_action_goal_event
 }
 
 // Transition map
-rcl_action_goal_event_handler
-  _goal_state_transition_map[GOAL_STATE_NUM_STATES][GOAL_EVENT_NUM_EVENTS] = {
-  [GOAL_STATE_ACCEPTED] = {
-    [GOAL_EVENT_EXECUTE] = _execute_event_handler,
-    [GOAL_EVENT_CANCEL] = _cancel_event_handler,
-  },
-  [GOAL_STATE_EXECUTING] = {
-    [GOAL_EVENT_CANCEL] = _cancel_event_handler,
-    [GOAL_EVENT_SET_SUCCEEDED] = _set_succeeded_event_handler,
-    [GOAL_EVENT_SET_ABORTED] = _set_aborted_event_handler,
-  },
-  [GOAL_STATE_CANCELING] = {
-    [GOAL_EVENT_SET_SUCCEEDED] = _set_succeeded_event_handler,
-    [GOAL_EVENT_SET_ABORTED] = _set_aborted_event_handler,
-    [GOAL_EVENT_SET_CANCELED] = _set_canceled_event_handler,
-  },
-};
+static rcl_action_goal_event_handler
+  _goal_state_transition_map[GOAL_STATE_NUM_STATES][GOAL_EVENT_NUM_EVENTS];
```

Comment:
@jacobperron just out of curiosity, what  was wrong about previous direct initialization?

Reply:
When the direct initialization gets included in C++ (e.g. gtest), then gcc complains about a sparsely populated 2d array. But we might be able to use null pointers in the direct initialization instead of this init function.

Reply:
Moving the direct implementation to the C file got rid of the compiler warnings :)